packages/protocol/contracts-0.8/common/MintGoldSchedule.sol

```
@@ -0,0 +1,297 @@
+    return (1, 1, 0, 0);
+  }
+
+  /**
+   * @notice Sets the community reward percentage
+   * @param value The percentage of the total reward to be sent to the community funds.
+   * @return True upon success.
+   */
+  function setCommunityRewardFraction(
+    uint256 value
+  ) public onlyOwner whenDependenciesSet returns (bool) {
+    uint256 timeSinceL2 = block.timestamp - l2StartTime;
+    uint256 linearSecondsLeft = SECONDS_LINEAR - (l2StartTime - genesisStartTime);
+
+    require(
+      timeSinceL2 < linearSecondsLeft,
+      "Can only update fraction once block reward calculation for years 15-30 has been implemented."
+    );
+    require(
+      value != communityRewardFraction.unwrap() && value < FixidityLib.fixed1().unwrap(),
+      "Value must be different from existing community reward fraction and less than 1."
+    );
+    communityRewardFraction = FixidityLib.wrap(value);
+    require(
+      FixidityLib.newFixed(1).gte(communityRewardFraction.add(carbonOffsettingFraction)),
+      "Sum of partner fractions must be less than or equal to 1."
+    );
+    emit CommunityRewardFractionSet(value);
+    return true;
+  }
+
+  /**
+   * @notice Sets the carbon offsetting fund.
+   * @param partner The address of the carbon offsetting partner.
+   * @param value The percentage of the total reward to be sent to the carbon offsetting partner.
+   * @return True upon success.
+   */
+  function setCarbonOffsettingFund(
+    address partner,
+    uint256 value
+  ) public onlyOwner whenDependenciesSet returns (bool) {
+    uint256 timeSinceL2 = block.timestamp - l2StartTime;
+    uint256 linearSecondsLeft = SECONDS_LINEAR - (l2StartTime - genesisStartTime);
+    require(
+      timeSinceL2 < linearSecondsLeft,
+      "Can only update fraction once block reward calculation for years 15-30 has been implemented."
+    );
+    require(partner != address(0), "Partner cannot be the zero address.");
```

This check could be moved to the beginning so we can fail fast and potentially save some gas